node-webservice/react-client/src/Components/TextInput.jsx

```diff
@@ -19,29 +19,43 @@ const submitStyle = {
 class TextInput extends Component {
     constructor(props) {
         super(props);
-        this.state = {value:''}
-        this.handleChange = this.handleChange.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
+        this.state = { nickname: "", gameID: "" }
 
-    handleChange(event){
-        this.setState({value: event.target.value.toUpperCase()})
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.isInputValid = this.isInputValid.bind(this);
     }
 
     handleSubmit(event){
         this.props.onSubmit( {
-            gameID: this.state.value,
-            username: "default_username" // TODO: add a text field in this form for username
+            gameID: this.state.gameID,
+            username: this.state.nickname
         });
         
         event.preventDefault();
     }
 
+    isInputValid() {
+        return this.state.gameID.length === 5 && this.state.nickname.length > 1;
```

Comment:
Do you mean for this to be > 1 or should it be > 0

Reply:
Hmm yeah i guess 1 character usernames should be allowed